examples/filesystem/main.c

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright (C) 2018 OTA keys S.A.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     examples
+ * @{
+ *
+ * @file
+ * @brief       File system usage example application
+ *
+ * @author      Vincent Dupont <vincent@otakeys.com>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <fcntl.h>
+
+#include "shell.h"
+#include "board.h" /* MTD_0 is defined in board.h */
+
+/* Flash mount point */
+#define FLASH_MOUNT_POINT   "/sda"
+
+#ifdef MTD_0
+/* File system descriptor initialization */
+#if defined(MODULE_LITTLEFS)
+#include "fs/littlefs_fs.h"
+
+static littlefs_desc_t fs_desc = {
+    .lock = MUTEX_INIT,
+};
+
+#define FS_DRIVER littlefs_file_system
+#elif defined(MODULE_SPIFFS)
```

Comment:
add some air around these `#if MODULE` pieces and a short comment heading on each of them, for helping users learning from this example. "Using LittleFS as the example file system" etc.